prototype/water-extent-map-on-demand.ipynb

```diff
@@ -0,0 +1,341 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "job_definition = {\n",
+    "    'name': 'water-map-example',\n",
+    "    'job_type': 'WATER_MAP',\n",
+    "    'job_parameters': {\n",
+    "        'resolution': 30,\n",
+    "        'speckle_filter': True,\n",
+    "        'max_vv_threshold': -15.5,\n",
+    "        'max_vh_threshold': -23.0,\n",
+    "        'hand_threshold': 15.0,\n",
+    "        'hand_fraction': 0.8,\n",
+    "        'membership_threshold': 0.45,\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2.3 Submit the jobs to the custom HyP3-watermap deployment\n",
+    "\n",
+    "Using the job definition as defined above (make sure you run the cell!), this will submit a job for\n",
+    "each granule in the search results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import copy\n",
+    "\n",
+    "prepared_jobs = []\n",
+    "for granule in all_granules:\n",
+    "    job = copy.deepcopy(job_definition)\n",
+    "    job['job_parameters']['granules'] = [granule]\n",
+    "    prepared_jobs.append(job)\n",
+    "\n",
+    "jobs = hyp3_watermap.submit_prepared_jobs(prepared_jobs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once the jobs are submitted, you can watch for them to complete (it will take ~30 min for all jobs to finish)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "jobs = hyp3_watermap.watch(jobs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Or, you can come back later and find your jobs by name, and make sure they're finished"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "jobs = hyp3_watermap.find_jobs(name='water-map-example')\n",
+    "jobs = hyp3_watermap.watch(jobs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once all jobs are complete, you can download the products for each successful job"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "jobs.download_files('data/')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Notes on viewing/evaluating the water map products\n",
+    "\n",
+    "* All GeoTIFFs in the RTC products are Cloud-Optimized, including the water map files `*_WM.tif`, and will have overviews/pyramids.\n",
+    "\n",
+    "  **This means the `*_WM.tif`'s appear to have a significantly higher water extent than they do in reality until you zoom in.**"
```

Comment:
We could explore alternate `OVERVIEW_RESAMPLING` settings when generating the WM COGs. I'm not sure if there are any other settings that might apply.
https://gdal.org/drivers/raster/cog.html#creation-options

Reply:
Nearest neighbor was specifically asked for (the default and what we're doing) -- so I think we're good here